crates/lox_core/src/time.rs

```diff
@@ -6,8 +6,85 @@
  * file, you can obtain one at https://mozilla.org/MPL/2.0/.
  */
 
+use crate::errors::LoxError;
+use crate::time::dates::Date;
+use crate::time::scales::TimeScale;
+use std::fmt::Display;
+
 pub mod constants;
+pub mod continuous;
 pub mod dates;
-pub mod epochs;
 pub mod intervals;
 pub mod leap_seconds;
+pub mod scales;
+pub mod utc;
+
+/// `WallClock` is the trait by which high-precision time representations expose human-readable time components.
+///
+/// The components returned by a `WallClock` must be interpreted in terms of its [TimeScale]. For example, a UTC
+/// `WallClock` will have a `second` component of 60 during a leap second.
+pub trait WallClock {
+    fn scale(&self) -> TimeScale;
+    fn hour(&self) -> u8;
+    fn minute(&self) -> u8;
+    fn second(&self) -> u8;
+    fn millisecond(&self) -> Thousandths;
+    fn microsecond(&self) -> Thousandths;
+    fn nanosecond(&self) -> Thousandths;
+    fn picosecond(&self) -> Thousandths;
+    fn femtosecond(&self) -> Thousandths;
+    fn attosecond(&self) -> Thousandths;
+}
+
+/// Newtype wrapper for thousandths of an SI-prefixed subsecond (milli, micro, nano, etc.).
+#[repr(transparent)]
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Thousandths(u16);
```

Comment:
That's a moutful and a real tongue twister for non-native speakers 😬 

How about [`Submultiple`](https://en.wikipedia.org/wiki/Metric_prefix) instead?

Reply:
How about `PerMille`?